### PR TITLE
feat(AWS SQS): Supports property scalingConfig

### PR DESCRIPTION
**Pull request by coderhglee**
Closes: #11676

### CODE REVIEW

**Comment by coderhglee** [comment]
Hi @medikoo 

I defined `MaximumConcurrency` as `ScalingConfig` object properties. This is a reference to the CloudFormation specification.

**CloudFormation Spec**
```
"MaximumConcurrency": {
            "description": "The maximum number of concurrent functions that an event source can invoke.",
            "type": "integer",
            "minimum": 2,
            "maximum": 1000
        },
        "ScalingConfig": {
            "description": "The scaling configuration for the event source.",
            "type": "object",
            "additionalProperties": false,
            "properties": {
                "MaximumConcurrency": {
                    "description": "The maximum number of concurrent functions that the event source can invoke.",
                    "$ref": "#/definitions/MaximumConcurrency"
                }
            }
        }
```

[SAM Template Ref](https://github.com/aws-samples/aws-lambda-amazon-sqs-max-concurrency/blob/main/template.yaml#L50)

If you think it will be duplicated with this [PR](https://github.com/serverless/serverless/pull/11678), this PR will be closed.

Thanks you.

File: lib/plugins/aws/package/compile/events/sqs.js [hunk 1]
```diff
@@ -111,6 +119,12 @@ class AwsCompileSQSEvents {
               };
             }
 
+            if (event.sqs.scalingConfig) {
+              sqsTemplate.Properties.ScalingConfig = {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/sqs.js:123]: @coderhglee, this doesn't seem correct CF config. Have you tested this configuration?

At least the AWS CF reference doesn't mention the `ScalingConfig` property -> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html.

I have a notion that you've got inspired by SAM configuration, which is a more high-level thing than CloudFormation we're relying on here.

**Review by medikoo — CHANGES_REQUESTED** [review]

File: lib/plugins/aws/package/compile/events/sqs.js (hunk 1 above)
Comment (coderhglee) [reply lib/plugins/aws/package/compile/events/sqs.js:123]: Yes I have done testing with this configuration and it deployed fine. 

The document you shared has not been updated.

You can check the CF config scheme `CloudFormation > Registry: Activated extensions > AWS::Lambda::EventSourceMapping`.

**CF stack**
```
 "some_resources": {
    "Type": "AWS::Lambda::EventSourceMapping",
    "Properties": {
      "BatchSize": 10,
      "MaximumBatchingWindowInSeconds": 10,
      "Enabled": true,
      "ScalingConfig": {
        "MaximumConcurrency": 20
      }
    }
  },
```

스크린샷 2023-01-16 오후 9 29 33

Comment (medikoo) [reply lib/plugins/aws/package/compile/events/sqs.js:123]: >  CloudFormation > Registry: Activated extensions > AWS::Lambda::EventSourceMapping.

@coderhglee can you pass a link to that?  Where exactly I can find this resource?

Comment (coderhglee) [reply lib/plugins/aws/package/compile/events/sqs.js:123]: @medikoo you can download Region resource provider schema files. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html 

`ScalingConfig.MaximumConcurrency` is defined in aws-lambda-eventsourcemapping.json file.

**Review by medikoo — COMMENTED** [review]
@coderhglee Thanks for the explanations.

I think it'll be better to make it a top-level property, aside of `maximumBatchingWindow,` as it's proposed in the original issue.

**Comment by coderhglee** [comment]
> I think it'll be better to make it a top-level property, aside of `maximumBatchingWindow,` as it's proposed in the original issue.

@medikoo Why do you think it's better to make it a top-level properties? What if another property is added to the Scaling config object in the future?

**Comment by coderhglee** [comment]
@medikoo Thanks for detail. It make sense 

So should I close this PR? I think it can be replaced by #11678. Or should I change it to top level property?

**Comment by medikoo** [comment]
Closing as it was addressed with https://github.com/serverless/serverless/pull/11678

**Pull request closed without merging**